xarray/core/computation.py

```diff
@@ -923,6 +923,12 @@ def apply_ufunc(
     Single value or tuple of Dataset, DataArray, Variable, dask.array.Array or
     numpy.ndarray, the first type on that list to appear on an input.
 
+    Notes
+    -----
+    This function is designed for the more common case where ``func`` can work on numpy
+    arrays. If ``func`` needs to manipulate a whole xarray object subset to each block
+    it is possible to use ``map_blocks``.
```

Comment:
do we need to state that it is recommended to use `apply_ufunc` wherever possible?

Reply:
sure.